samcli/hook_packages/terraform/prepare/main.bat

```diff
@@ -0,0 +1,14 @@
+:: ensure the script works regardless of where it's invoked from
+@echo off&setlocal
+for %%i in ("%~dp0.") do set "parent_path=%%~fi"
+cd %parent_path%
+
+:: set up virtual environment
+python -m venv ./venv
```

Comment:
- do we need to check if the virtual environment is already there?
- do we need to check the python is installed and return errors if it is not installed?
- we need to make sure that these commands are not logging to the output stream, we need to forward it to some logs file. The idea is the wrapper will read the prepare hook output from the command output stream.

Reply:
I also have some questions:
- If we use `./venv` here, are we assuming the script runs in the current project root? 
- What if the customer already has a virtual env of the same name created for a different purpose? 
- if we are re-using an existing virtual env and there's a version conflict in one of the dependencies, how shall we handle it?

Reply:
- nope, the script runs in the `prepare` directory always (check out the `ensure the script works regardless of where it's invoked from` line)
- not a problem due to above ^

Reply:
do we want to create a virtual env in the `prepare` directory? Since it will live under where the SAM CLI is installed. I supposed this should be touched only by Homebrew or any other package manager like pip.

For example, homebrew installed my SAM CLI at `/usr/local/Cellar/aws-sam-cli/1.52.0/libexec/lib/python3.8/site-packages`. So if we create the venv in the `prepare` directory, it will be `/usr/local/Cellar/aws-sam-cli/1.52.0/libexec/lib/python3.8/site-packages/samcli/hook_packages/terraform/prepare/venv`. I'm not sure if we should touch anything under `/usr/local/Cellar/aws-sam-cli/1.52.0/libexec/lib/python3.8/site-packages`